wrappers/services.go

```diff
@@ -114,24 +114,26 @@ func StartServices(apps []*snap.AppInfo, inter interacter) (err error) {
 			if err == nil {
 				return
 			}
-			if e := stopService(sysd, app, inter); e != nil {
-				inter.Notify(fmt.Sprintf("While trying to stop previously started service %q: %v", app.ServiceName(), e))
+			if app.ServiceMode() == systemd.SystemMode {
```

Comment:
I think you are doing the right thing here (ie, not mixing in user services with the `snap start|stop|restart` commands, but it makes me wonder if `snap start|stop|restart` should grow a `--user` option or similar for these...

Reply:
It's not something that snapd can control though: in general snapd running as pid 0 is not going to be able to communicate with the systemd instance running as the user.

At the moment we don't do anything about snap apps running as the user during an upgrade, so I don't think it is that big a stretch to do the same for user services.

Reply:
Note that this is something that (iirc), @chipaca is looking into. It would be possible for snapd to communicate with userd to do something sensible on refresh. Also note that my question is not meant to block this PR; just putting it out there for a potential followup.